### ISSUE TITLE
Digital Ocean Droplet creation failed with DoError: found unpermitted parameters: virtio

### ISSUE THREAD

**Issue by boberhaus** [trailofbits/algo#1571]
Perhaps DO API has changed? Error output:

TASK [cloud-digitalocean : Creating a droplet...] ******************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: DoError: found unpermitted parameters: virtio
fatal: [localhost]: FAILED! => {"changed": false, "msg": "found unpermitted parameters: virtio"}
included: /home/beoberha/algo/playbooks/rescue.yml for localhost

TASK [debug] *******************************************************************
ok: [localhost] => {
    "fail_hint": [
        "Sorry, but something went wrong!",
        "Please check the troubleshooting guide.",
        "https://trailofbits.github.io/algo/troubleshooting.html"
    ]
}

TASK [Fail the installation] ***************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Failed as requested from task"}

PLAY RECAP *********************************************************************
localhost                  : ok=32   changed=6    unreachable=0    failed=2

**Comment by cmersinli** [comment]
i've same issue

**Comment by boberhaus** [comment]
It seems that the [ansible module algo uses to create droplets](
https://github.com/trailofbits/algo/blob/25513cf9259646fc2207869fe853c74b7355e4a3/roles/cloud-digitalocean/tasks/main.yml#L54) [is deprecated](https://github.com/ansible/ansible/blob/63f7536594ce43cfe015202adf84d6c90e1c99ba/lib/ansible/modules/cloud/digital_ocean/_digital_ocean.py#L20)

[It uses virtio as a default param](https://github.com/ansible/ansible/blob/63f7536594ce43cfe015202adf84d6c90e1c99ba/lib/ansible/modules/cloud/digital_ocean/_digital_ocean.py#L69) 

[However, the new module does not have a virtio param](https://github.com/ansible/ansible/blob/63f7536594ce43cfe015202adf84d6c90e1c99ba/lib/ansible/modules/cloud/digital_ocean/digital_ocean_droplet.py#L17) so I guess that the API no longer accepts it. 

I will play around with a fix when I have some time.

**Comment by TC1977** [comment]
Fix for that is in #1549. Try installing that branch.

**Comment by davidemyers** [comment]
@TC1977 Actually #1549 is failing for me now too with a different but similar error:
```
found unpermitted parameters: id
```

**Comment by davidemyers** [comment]
Jack has added a fix for this issue to PR #1549 and also submitted a fix to Ansible. If you need to deploy to DigitalOcean you can help test out the code in the PR, which now uses Python 3 instead of Python 2.

Get a copy of the branch like this:
```
git clone -b ansible-2.8 https://github.com/trailofbits/algo.git
```
Be sure to read and follow the setup instructions in the revised [README](https://github.com/trailofbits/algo/blob/ansible-2.8/README.md) due to the change to Python 3.

**Comment by andrewsomething** [comment]
I'd like to apologize for the disruption. This was raised with the team at DigitalOcean. There had been a change deployed to the /v2/droplets endpoint to reject requests with extra, non-supported parameters. This was not communicated properly, and broke existing implementations. The change has been rolled back, and this should now work as it did previously. Please let me know if you experience more issues.
